tests/phpunit/tests/image/functions.php

```diff
@@ -726,6 +726,68 @@ public function test_wp_generate_attachment_metadata_pdf() {
 		remove_filter( 'wp_upload_image_mime_transforms', '__return_empty_array' );
 	}
 
+	/**
+	 * @ticket 55443
+	 */
+	public function test_wp_generate_attachment_metadata_pdf_webp_mime() {
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'application/pdf' ) ) ) {
+			$this->markTestSkipped( 'Rendering PDFs is not supported on this system.' );
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'image/webp' ) ) ) {
+			$this->markTestSkipped( 'This test requires WebP support.' );
+		}
+
+		// Use legacy JPEG output and WebP.
+		add_filter(
+			'wp_upload_image_mime_transforms',
+			function () {
+				return array(
+					'image/jpeg'      => array( 'image/jpeg', 'image/webp' ),
+					'image/webp'      => array( 'image/webp', 'image/jpeg' ),
+					'application/pdf' => array( 'image/webp', 'image/jpeg' ),
+				);
+			}
+		);
+
+		$orig_file = DIR_TESTDATA . '/images/wordpress-gsoc-flyer.pdf';
+		$test_file = get_temp_dir() . 'wordpress-gsoc-flyer.pdf';
+		copy( $orig_file, $test_file );
```

Comment:
why does this need to make a copy of the test file, can it use the test file directly?

Reply:
@adamsilverstein actually we need the file to be in temp directory to be able to create the upload object for the attachment and later to delete it.